### ISSUE TITLE
Silently crashes neovim with elvis-operator null-safety handling

### ISSUE THREAD

**Issue by covelloz** [fwcd/tree-sitter-kotlin#134]
I'm noticing that the kotlin tree-sitter extension is crashing neovim when trying to throw errors on nullable types. I will give a very simple, reproducible example. I have confirmed that the issue is with tree-sitter-kotlin. If I uninstall kotlin from tree-sitter the issue resolves: `:TSUninstall kotlin`

tree-sitter: current ([v0.9.2](https://github.com/nvim-treesitter/nvim-treesitter/releases/tag/v0.9.2))
tree-sitter-kotlin: current ([0.3.7](https://github.com/fwcd/tree-sitter-kotlin/releases/tag/0.3.7))
Neovim version:
```
NVIM v0.10.1
Build type: Release
LuaJIT 2.1.1720049189
Run "nvim -V1 -v" for more info
```

Example:
```
package org.example

import kotlin.random.Random

class App {
    val greeting: String
        get() {
            val rng = Random.Default.nextInt(1, 11)
            val maybeSmInt: Int? = if (rng > 5) rng else null
            val safeSmlInt: Int = maybeInt ?: throw RuntimeException("not a sm int!!") <-- tree-sitter-kolin crashes here when trying to provide/type out an error message
            return "Hello World!"
        }
}

fun main() {
    println(App().greeting)
}
```

https://github.com/user-attachments/assets/0153779f-1df3-4c2f-944a-2fdd35cfcb96

**Comment by fwcd** [comment]
This repo only contains the grammar, so there is very little surface for things to crash outside of maybe the external scanner. Your example seems to work fine in the [web playground](https://fwcd.dev/tree-sitter-kotlin/). Have you tried filing the issue with the devs of the Neovim extension that integrates tree-sitter-kotlin?

**Comment by covelloz** [comment]
@fwcd it very well could be [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) itself. Given that it works in the web playground - I would assume you're probably correct. I will close this & post it over there. Thanks for the sanity check!

**Comment by fwcd** [comment]
Perhaps it was the external scanner after all, #136 fixed a segfault in the deserialization logic. Could you check whether tree-sitter-kotlin 0.3.8 fixes the issue for you?

**Comment by covelloz** [comment]
@fwcd I can confirm that the issue is no longer happening after running `:TSUpdate` & updating the Kotlin parser. Looks like issue is fixed as of 0.3.8!
